### PR TITLE
Added os-net-http-avoid-escapes... patch to 1.15

### LINKED ISSUE #673: os, net/http: avoid escapes from os.DirFS and http.Dir on Windows CVE-2022-41720 - [eks go1.15 backport]
A backport to EKS Go 1.15 is [requested by markapruett](https://github.com/aws/eks-distro-build-tooling/issues/671#issuecomment-1341499171)

### PR DESCRIPTION
**Pull request by kschumy**
*Issue #, if available:*
closes #673

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/golang/go/1.15/patches/0101-Don-t-use-the-bundled-tzdata-at-runtime-except-for-t.patch [hunk 1]
```diff
@@ -1,4 +1,4 @@
-From edce31a2904846ae74e3c011f2cf5fddc963459e Mon Sep 17 00:00:00 2001
```
Comment (jaxesn) [inline projects/golang/go/1.15/patches/0101-Don-t-use-the-bundled-tzdata-at-runtime-except-for-t.patch:1]: Can we not change these? I am assuming its a side effect of the way we apply the patches triggers the shas to change?

File: projects/golang/go/1.15/patches/0023-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch [hunk 2]
```diff
@@ -0,0 +1,394 @@
+From ecd61a3d6483d032aea259c2b2ba7e8f34f10c8f Mon Sep 17 00:00:00 2001
+From: Damien Neil <dneil@google.com>
+Date: Thu, 10 Nov 2022 12:16:27 -0800
+Subject: [PATCH] [go-1.15.15-eks] os, net/http: avoid escapes from
+ os.DirFS and http.Dir on Windows
+
+Do not permit access to Windows reserved device names (NUL, COM1, etc.)
```
Comment (kschumy) [inline projects/golang/go/1.15/patches/0023-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch:7]:
```suggestion
# AWS EKS 
Backported To: go-1.15.15-eks
Backported On: Fri, 16 Dec 2022
Backported By: ksschumy@amazon.com
Backported From: release-branch.go1.18
Upstream Source Commit: https://github.com/golang/go/commit/7013a4f5f816af62033ad63dd06b77c30d7a62a7

# Original Information

Do not permit access to Windows reserved device names (NUL, COM1, etc.)
```

File: projects/golang/go/1.15/rpmbuild/SPECS/golang.spec [hunk 3]
```diff
@@ -246,6 +246,7 @@ Patch19:       0019-go-1.15.15-eks-path-filepath-fix-stack-exhaus.patch
 Patch20:       0020-go-1.15.15-eks-net-http-update-bundled-golang.patch
 Patch21:       0021-add-method-to-skip-privd-tests-if-required.patch
 Patch22:       0022-go-1.15.15-eks-archive-tar-limit-size-of-head.patch
+Patch23:       0022-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch
```
Comment (kschumy) [inline projects/golang/go/1.15/rpmbuild/SPECS/golang.spec:249]:
```suggestion
Patch23:       0023-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch
```

File: projects/golang/go/1.15/patches/0023-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+new file mode 100644
+index 0000000000..f93da18680
+--- /dev/null
++++ b/src/internal/safefilepath/path_other.go
+@@ -0,0 +1,23 @@
++// Copyright 2022 The Go Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style
++// license that can be found in the LICENSE file.
++
++//go:build !windows
```
Comment (danbudris) [inline projects/golang/go/1.15/patches/0023-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch:105]:
```suggestion
+// +build !windows
```

File: projects/golang/go/1.15/patches/0023-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+@@ -0,0 +1,88 @@
++// Copyright 2022 The Go Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style
++// license that can be found in the LICENSE file.
++
++package safefilepath_test
++
++import (
++	"internal/safefilepath"
++	"ioutil"
```
Comment (rcrozean) [inline projects/golang/go/1.15/patches/0023-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch:138]:
```suggestion
+	"io/ioutil"
```

Comment (danbudris) [inline projects/golang/go/1.15/patches/0023-go-1.15.15-eks-os-net-http-avoid-escapes-from-os.Dir.patch:138]:
```suggestion
+	"io/ioutil"
```

**Pull request merged**